docs/04-operating-scs/08-backups/user_data_backups.md

```diff
@@ -0,0 +1,303 @@
+Before downloading secrets from Barbican make sure that a secure target environment is established for receiving and securely storing the secret's contents.
+
+:::
+
+Barbican secrets can be downloaded in plaintext using the corresponding API or client command:
+
+```bash
+openstack secret get --file $TARGET_FILE_PATH --payload_content_type "application/octet-stream" $SECRET_ID
+```
+
+### Retrieving encryption keys from Barbican
+
+In case of encrypted volumes (i.e. using a volume type with encryption), a corresponding encryption key is stored in Barbican.
+When an image is created from such a volume, the encryption key is duplicated in Barbican for the image.
+In order to backup those keys, the corresponding secret must first be identified.
+
+For volumes, this is possible starting with the Volume API microversion 3.64:
+
+```bash
+openstack volume show --os-volume-api-version 3.64 $VOLUME_NAME_OR_ID
+```
+
+The response will contain an `encryption_key_id` field with the ID of the Barbican secret.
+
+For images, the secret reference is stored in the `properties` field instead:
+
+```bash
+openstack image show -f value -c properties $IMAGE_NAME_OR_ID
+```
+
+In case of images created from encrypted volumes, the resulting output will have a nested `cinder_encryption_key_id` field that contains the ID of the Barbican secret.
+
+The resulting IDs can be used to retrieve the corresponding key using the [Barbican instructions](#barbican-secrets-backup-using-download) above.
+
+:::caution
+
+Note that the key retrieved from the secret is not immediately usable as LUKS passphrase to the image data of the volume.
+OpenStack does some processing to the key before it is passed to the LUKS encryption, which must be mimicked accordingly in order to unlock the encryption outside of OpenStack!
+
+See the [example procedure for converting the LUKS key](#luks-encryption-key-conversion-to-decrypt-volume-images) in the appendix section.
+
+:::
+
+## Appendix
+
+### Image creation action for servers with attached volumes
+
+When the `createImage` action of the Compute API (`openstack server image create`) is used on virtual machines that have at least one volume attached, a snapshot will be created for each attached volume individually and referenced in the resulting image's metadata.
+
+This happens regardless of whether the virtual machine has an Ephemeral Storage disk attached.
+In case of an Ephemeral Storage disk, only the Ephemeral Storage is copied into the Glance image as a 1:1 copy.
+
+In case of a virtual machine that has no Ephemeral Storage but only volumes, the `createImage` action leads to a Glance image that only consists of metadata (including the resulting volume snapshot references) but carries no actual binary data.
+
+![Figure: createImage action flow involving Ephemeral Storage and/or volumes](./images/user_data_backups_figure1.png)
```

Comment:
Why is the shelve command integrated there?
For ephemeral storage the procedure is alike, but I don't know whether the metadata of the image is the same for both.
In case of the volume it will not be touched for the shelve action.

Reply:
Good point. Maybe I need to be a bit more verbose about the shelve operation in the rest of the documentation. Shelve could be considered some form of (at least temporary) backup from the user perspective. Since the same (imho) non-obvious behavior as for `server image create` regarding mixed ephemeral and volume storage applies for shelve, I wanted to be as clear as possible in the user guide.

Reply:
Actually, you are right. I checked again and the behavior differs from `image create`. No snapshots are taken of volumes and no image is created if there is no Ephemeral Storage. I don't think shelve could count as any form of backup after all.

I have removed the references to shelve from the figure. Thanks for bringing this up!